### PR TITLE
[cleanup] consistent __init__.py for import *

### LINKED ISSUE #471: [dev] should we support import *?
During this code review, this question came up. Thanks to @msbaines !

https://github.com/facebookresearch/fairscale/pull/467#discussion_r587572010

The question is around whether we should support `import *` by maintaining `__all__` in the `__init__.py` file.

cc: @anj-s @blefaudeux

### PR DESCRIPTION
**Pull request by min-xu-ai**
- fixes #471
- one less thing to worry about during development.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by blefaudeux — APPROVED** [review]
LGTM, thanks Min for the careful changes and wording all around

File: fairscale/nn/__init__.py [hunk 1]
```diff
@@ -3,23 +3,12 @@
...
 
+from typing import List
+
 from .data_parallel import FullyShardedDataParallel, ShardedDataParallel
 from .misc import FlattenParamsWrapper, checkpoint_wrapper
 from .moe import MOELayer, Top2Gate
 from .pipe import Pipe, PipeRPCWrapper
 from .wrap import auto_wrap, default_auto_wrap_policy, enable_wrap, wrap
 
-__all__ = [
```
Comment (anj-s) [inline fairscale/nn/__init__.py:12]: just curious, what prevents existing import statements from breaking if we remove this?

**Review by anj-s — APPROVED** [review]
thanks @min-xu-ai for this change!

File: fairscale/nn/__init__.py (hunk 1 above)
Comment (min-xu-ai) [reply fairscale/nn/__init__.py:12]: Nothing breaks. It simply changes the behavior of "import *", which is only used during debugging session etc. afaik.

**Pull request merged**